packages/Parallel/WorkerRunner.php

```diff
@@ -129,23 +131,24 @@ public function run(Encoder $encoder, Decoder $decoder, Configuration $configura
 
     /**
      * @param array{system_errors: SystemError[], file_diffs: FileDiff[]}|mixed[] $errorAndFileDiffs
-     * @return array{system_errors: SystemError[], file_diffs: FileDiff[]}
      */
-    private function processFile(File $file, Configuration $configuration, array $errorAndFileDiffs): array
+    private function processFile(File $file, Configuration $configuration, array $errorAndFileDiffs): ProcessFileResult
     {
+        $wasFileChanged = false;
         foreach ($this->fileProcessors as $fileProcessor) {
             if (! $fileProcessor->supports($file, $configuration)) {
                 continue;
             }
 
             $currentErrorsAndFileDiffs = $fileProcessor->process($file, $configuration);
+            $wasFileChanged = $wasFileChanged || $currentErrorsAndFileDiffs[Bridge::FILE_DIFFS] !== [];
```

Comment:
Defensive `||`, if file was changed even in just one file processor, do not cache it.